logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/auth/LoginUseCase.kt

```diff
@@ -133,16 +136,23 @@ internal class LoginUseCaseImpl internal constructor(
             .fold({
                 when (it) {
                     is NetworkFailure.ProxyError -> AuthenticationResult.Failure.SocketError
-                    is NetworkFailure.ServerMiscommunication -> handleServerMiscommunication(it)
+                    is NetworkFailure.ServerMiscommunication -> handleServerMiscommunication(it, isEmail, cleanUserIdentifier)
                     is NetworkFailure.NoNetworkConnection, NetworkFailure.FederatedBackendFailure ->
                         AuthenticationResult.Failure.Generic(it)
                 }
             }, {
+                if (isEmail && secondFactorVerificationCode != null) {
```

Comment:
maybe an overkill but maybe we can move it to a temp variable with more clear indication when we are storing the verification code ? like what is the reason we are storing the verification code here ? I assume it is because the state allows us to do so, but definetly a minor think